lib/active_admin/filters/humanized.rb

```diff
@@ -9,8 +9,22 @@ def initialize(param)
         @value = param[1]
       end
 
+      def related_class
+        @related_class ||=
+            begin
+              @body.split('_id').first.gsub(' ', '').classify.constantize
+            rescue
+              nil
+            end
+      end
+
       def value
-        @value
+        @value =
+            begin
+              display_name(related_class.find(@value))
```

Comment:
@Shoroh That's a good point, I think your code is good as-is then. Would probably want to check with @timoschilling though.